scripts/main.py

```diff
@@ -0,0 +1,150 @@
+import os
+from dotenv import load_dotenv
+import tweepy as tw
+import re
+import string
+import nltk
+import pandas as pd
+from sklearn.cluster import DBSCAN
+from sklearn.feature_extraction.text import TfidfVectorizer
+
+# NLTK variables to aid in cleaning/preprocessing
+ps = nltk.PorterStemmer()
+
+# Pandas dataframe for tweets
+df = pd.DataFrame(columns=["id", "tweet", "popularity"])
```

Comment:
Avoid using global declaration, try initializing it in the inside the class or function and you can pass it as the parameter wherever required.

Reply:
Alright, I will move the `ps = nltk.PorterStemmer()` to the function. However, would it not be more costly to pass the `df` as a parameter everywhere? It is used virtually everywhere in this code.

Reply:
Also, adding on to the port stemmer topic, would it not be better to have it as a global variable because it is called multiple times in the tokenizer which I have no control over passing the parameters to.

Reply:
This might be used at most of the places but still not at all the places and they contain information, which are subject to override so putting them inside the class will yield a better result.

Also, it will get re-initialized multiple times whenever the file is imported.

Reply:
Alright. Got it!